crates/bevy_sprite/src/render/mod.rs

```diff
@@ -237,19 +238,68 @@ pub fn extract_sprites(
         if !visibility.is_visible {
             continue;
         }
-        // PERF: we don't check in this function that the `Image` asset is ready, since it should be in most cases and hashing the handle is expensive
-        extracted_sprites.sprites.alloc().init(ExtractedSprite {
-            color: sprite.color,
-            transform: *transform,
-            // Use the full texture
-            rect: None,
-            // Pass the custom size
-            custom_size: sprite.custom_size,
-            flip_x: sprite.flip_x,
-            flip_y: sprite.flip_y,
-            image_handle_id: handle.id,
-            anchor: sprite.anchor.as_vec(),
-        });
+        if let SpriteDrawMode::Simple = sprite.draw_mode {
+            // PERF: we don't check in this function that the `Image` asset is ready, since it should be in most cases and hashing the handle is expensive
+            extracted_sprites.sprites.alloc().init(ExtractedSprite {
+                color: sprite.color,
+                transform: *transform,
+                // Use the full texture
+                rect: None,
+                custom_size: sprite.custom_size,
+                flip_x: sprite.flip_x,
+                flip_y: sprite.flip_y,
+                image_handle_id: handle.id,
+                anchor: sprite.anchor.as_vec(),
+            });
+            continue;
+        }
+        let image_size = match images.get(handle) {
+            None => continue,
+            Some(i) => Vec2::new(
+                i.texture_descriptor.size.width as f32,
+                i.texture_descriptor.size.height as f32,
+            ),
+        };
+
+        let slices = match &sprite.draw_mode {
+            SpriteDrawMode::Sliced(slicer) => slicer.compute_slices(
```

Comment:
Note though that if it doesn't take any significant amount of time for a lot of such entities (i.e. doesn't add more than say 0.1ms compared to as many entities that just use simple sprites) but it would take more time in a separate `prepare` stage system due to having to iterate over them all again, then it can be OK to do it here. So basically profile and present results.

Reply:
Having now seen the slicer code, I suggest moving that to the `prepare` stage.

Reply:
I'll move every computation in `prepare` once https://github.com/bevyengine/bevy/pull/5247 is done